core/src/main/scala/kafka/server/RaftServer.scala

```diff
@@ -43,7 +43,8 @@ import org.apache.kafka.raft.{FileBasedStateStore, KafkaRaftClient, QuorumState,
 
 import scala.jdk.CollectionConverters._
 
-class RaftServer(val config: KafkaConfig) extends Logging {
+class RaftServer(val config: KafkaConfig,
+                 val selfIncrement: Boolean) extends Logging {
```

Comment:
If we change the state machine, we can get rid of this. I think it's ok if our tests always rely on a separate producer to write data.